### PR TITLE
 Fixes #2017 - Change ID from query parameter to path segment 

### LINKED ISSUE #2017: URL Issue: ID Passed as Query Parameter Instead of Path Segment
**Describe the bug**

The URL currently uses a query parameter to pass the ID (e.g., url/id=343434) instead of including the ID as part of the path (e.g., url/343434). This format does not adhere to the expected RESTful URL structure.

**To Reproduce**

1. Login as a the user 
2. go the donation page 
3. click on "view all" on the right
4. you will see not any members because of the invalid url but if you remove " id=" from the url , you will see all members. 

**Expected behavior**
All members should be visible. 

**Actual behavior**
Members are not visible .

**Screenshots**

image

https://github.com/PalisadoesFoundation/talawa-admin/assets/71646466/e7df0612-30e5-4c59-8a23-6df88fc0da79

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by lokeshsingh3721**
**What kind of change does this PR introduce?**

This PR introduces a bug fix that changes the way IDs are passed in the URL. Previously, IDs were passed as query parameters, and now they will be passed as path segments. This change improves the readability and structure of the URLs.

**Issue Number:**

Fixes #2017 

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa-admin/assets/71646466/56b939e6-446a-4d17-8f56-16c47188a34f

**Does this PR introduce a breaking change?**

NO

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

YES

## Summary by CodeRabbit

- **Refactor**
	- Updated state initialization in the organization sidebar for better type handling.
	- Simplified URL structures by removing `id=` across various navigation links.

- **Bug Fixes**
	- Corrected URL paths in multiple test files to match the new simplified format. 

- **Tests**
	- Adjusted test cases to reflect changes in URL structures.

### CODE REVIEW

File: src/components/UserPortal/OrganizationSidebar/OrganizationSidebar.tsx [hunk 1]
```diff
@@ -77,32 +81,28 @@ export default function organizationSidebar(): JSX.Element {
...
+              return (
+                <ListGroup.Item
+                  key={member._id}
+                  action
+                  className={`${styles.rounded} ${styles.colorLight} my-1`}
+                >
+                  <div className="d-flex flex-row">
+                    <img
+                      src={member.image ? member.image : AboutImg}
+                      alt={`Image of ${memberName}`}
```
[the remark this answers is not in the document]
Comment (lokeshsingh3721) [reply src/components/UserPortal/OrganizationSidebar/OrganizationSidebar.tsx]: already done this change .

**Comment by lokeshsingh3721** [comment]
any update ?

**Comment by palisadoes** [comment]
Please fix the conflicting file and ensure that the changes requested by code rabbit are approved automatically by the code rabbit bot.

**Comment by lokeshsingh3721** [comment]
image
Locally this test is getting pass .

**Comment by palisadoes** [comment]
Something is different with your setup. Please fix the failing test (s)

**Comment by lokeshsingh3721** [comment]
> Something is different with your setup. Please fix the failing test (s)

 can I  re-run this test here?

**Comment by palisadoes** [comment]
Please make a minor commit to the same PR and the tests will run again automatically

**Comment by lokeshsingh3721** [comment]
@palisadoes  all checks are passed

**Comment by lokeshsingh3721** [comment]
@palisadoes  why is it asking for workflow approval?

**Comment by palisadoes** [comment]
> @palisadoes why is it asking for workflow approval?

This video explains our PR process and has many answers to questions

- https://youtu.be/dQuEKpsy1vM

**Comment by palisadoes** [comment]
Please fix the failing workflow / tests

**Pull request closed without merging**